src/liquidator.ts

```diff
@@ -291,6 +265,98 @@ async function main() {
   }
 }
 
+async function maybeLiquidateAccount(mangoAccount: MangoAccount): Promise<boolean> {
+    const mangoAccountKeyString = mangoAccount.publicKey.toBase58();
+
+    if (!mangoAccount.isLiquidatable(mangoGroup, cache)) {
+      console.log(
+        `Account ${mangoAccountKeyString} no longer liquidatable`,
+      );
+      return false;
+    }
+
+    const health = mangoAccount.getHealthRatio(mangoGroup, cache, 'Maint');
+    const accountInfoString = mangoAccount.toPrettyString(
+      groupIds,
+      mangoGroup,
+      cache,
+    );
+    console.log(
+      `Sick account ${mangoAccountKeyString} health ratio: ${health.toString()}\n${accountInfoString}`,
+    );
+    notify(`Sick account\n${accountInfoString}`);
+    try {
+      await liquidateAccount(
+        mangoGroup,
+        cache,
+        spotMarkets,
+        rootBanks,
+        perpMarkets,
+        mangoAccount,
+        liqorMangoAccount,
+      );
+
+      console.log('Liquidated account', mangoAccountKeyString);
+      notify(`Liquidated account ${mangoAccountKeyString}`);
+    } catch (err: any) {
+      console.error(
+        `Failed to liquidate account ${mangoAccountKeyString}: ${err}`,
+      );
+      notify(
+        `Failed to liquidate account ${mangoAccountKeyString}: ${err}`,
+      );
+    }
+
+    return true;
+}
+
+async function newAccountOnLiquidatableFeed(params) {
+  console.log(`Checking health of Account ${params.account}...`);
+  try {
+    const mangoAccountKey = new PublicKey(params.account);
+    const mangoAccount = new MangoAccount(mangoAccountKey, null);
+
+    [cache, liqorMangoAccount, ] = await Promise.all([
+      mangoGroup.loadCache(connection),
+      liqorMangoAccount.reload(connection, mangoGroup.dexProgramId),
+      mangoAccount.reload(connection, mangoGroup.dexProgramId),
+    ]);
+
+    const liquidated = await maybeLiquidateAccount(mangoAccount);
+    if (liquidated) {
+      cache = await mangoGroup.loadCache(connection);
+      await liqorMangoAccount.reload(connection, mangoGroup.dexProgramId);
+
+      // Check need to rebalance again after checking accounts
+      await balanceAccount(
+        mangoGroup,
+        liqorMangoAccount,
+        cache,
+        spotMarkets,
+        perpMarkets,
+      );
+    }
+  } catch (err) {
+    console.error('Error liquidating account:', err);
+  }
+}
+
+// never returns
+async function liquidatableFromLiquidatableFeed() {
+  const ws = new RpcWebSocketClient(liquidatableFeedWebsocketAddress, {
+    max_reconnects: Infinity,
```

Comment:
what happens when your remove websocket server dies, should you switch to the default mode?

Reply:
I assume the websocket server will restart and then we'll want to reconnect to it. I don't want to build fallover behavior.